src/server/src/proxy.ts

```diff
@@ -384,48 +373,51 @@ app.use(Express.json())
 app.use(Express.static('static'))
 
 // Define the number of proxy hops on the system to detect correct source IP for the filters below
-if (proxy_hops > 0) {
-  app.set('trust proxy', proxy_hops)
+if (settings.proxy_hops > 0) {
+  app.set('trust proxy', settings.proxy_hops)
 }
 
 // Set up blacklist and use the proxy number defined in the settings. Log only IP if blocked
-if (use_ip_blacklist) {
-  app.use(IpFilter(ip_blacklist, {logLevel: 'deny', trustProxy: proxy_hops}))
+if (settings.use_ip_blacklist) {
+  app.use(IpFilter(settings.ip_blacklist, {logLevel: 'deny', trustProxy: settings.proxy_hops}))
 }
 
 // Error handling
-app.use((err, req, res, _next) => {
+app.use((err: Error, req: Request, res: Response, _next: any) => {
   if (err instanceof IpDeniedError) {
     return res.status(401).json({error: 'IP has been blocked'})
   }
   else {
+    // @ts-ignore status field does not exist, only return err here?
     return res.status(500).json({error: err.status})
   }
 })
 
 // Define authentication service
-if (use_auth) {
+if (settings.use_auth) {
   app.use(BasicAuth({ authorizer: myAuthorizer }))
 }
 
 // Block IP if requesting too much but skipped if a valid token_key is provided (long interval)
-if (use_rate_limiter) {
+if (settings.use_rate_limiter) {
   const limiter1 = new RateLimiterMemory({
     keyPrefix: 'limit1',
-    points: rate_limiter.request_limit, // limit each IP to x requests per duration
```

Comment:
Moved some of these types of comments to the type declaration. Most IDEs supports reading that type signature as code documentation.